flink-external-resource/flink-external-resource-gpu/src/main/resources/gpu-discovery.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+if [ $# -lt 1 ]; then
+  echo "Usage: ./gpu-discovery-script.sh gpu-amount [--privilege] [--check-dead] [--assign-file filePath]"
+  exit 1
+fi
+
+
+AMOUNT=$1
+shift
+ASSIGNMENT_FILE="/var/tmp/flink-gpu-assignment"
+
+while [[ $# -ge 1 ]]
+do
+key="$1"
+  case $key in
+    --privilege)
+    PRIVILEGE_OPTION="privilege"
```

Comment:
It would be better to give it an empty default value before the loop. This prevents accidentally taking an environment variable with the same name.
Same for `CHECK_DEAD_PROCESS`.